pyproject.toml

```diff
@@ -31,6 +31,7 @@ dependencies = [
     "pandas>=2.2.2",
     "polars[pandas]<=1.17.0",
     "scikit-learn>=1.5.0",
+    "scipy>=1.15.1",
```

Comment:
As per the issue description, we should not use the latest scipy version. The version should be as high as needed (securitz fixes or major bugs) and as low as possible (compatibility). Based on my intuition I'd probably go with something from version 1.6 or upwards. But only intuition wise, please check changelogs etc.